src/you_get/extractors/youku.py

```diff
@@ -254,6 +278,7 @@ def extract(self, **kwargs):
                     streamfileid = piece['fileid']
                     for no in range(0, len(segs)):
                         k = segs[no]['key']
+                        assert k != -1
```

Comment:
Ok, I think I just figured this out:

`k != -1` could happen when the rest of video pieces are behind a paywall (VIP-only), we should just break here, instead of throwing an error, otherwise it would make such videos not downloadable at all (as in #996).

However, I'm not sure if changing this would fix #996 completely. It seems that the `play.youku.com` API still returns `"key":-1` even if users passed their own cookies. (though I'm NOT able to confirm this myself)